src/content/plugins/normal-module-replacement-plugin.md

```diff
@@ -40,50 +41,78 @@ Conditional build depending on an [specified environment](/configuration/configu
 Say you want a configuration with specific values for different build targets.
 
 ``` javascript
-module.exports = function(env) {
-  var appTarget = env.APP_TARGET || 'VERSION_A';
+module.exports = function(env, args) {
```

Comment:
`args`?

Reply:
See the last code section on https://webpack.js.org/concepts/mode/#mode-production. It appears that the signature and use case has changed from earlier versions of Webpack, and some of the documentation throughout may not have been updated to match the change. 

Based on the new signature, from a DevOps perspective, `env` would specify the environment that you are **in**, and `args` (I believe `argv` is a typo) would be any additional configuration you pass to the build. Any other values that you needed that are *specific* to the environment would either be built *into the script itself based on the environment name*, or passed in through the `args`. **The way that `mode` shows up as a property of `args` and not `env` in this example (and I have confirmed this behavior in my own apps) appears to back up my observation on this change.**

Based on the fact that passing `args` is even *possible*, I would recommend updating **all** of the documentation to make sure the signature is consistent for all Webpack users moving forward.

Reply:
CC @evenstensberg

Reply:
I should also point out that, if you do it this way, you get super clean and easy to understand build commands per environment. Consider the following elements in my Visual Studio 2017 project file:
```
    <Exec Command="npx webpack --config webpack.config.vendor.js --mode production" />
    <Exec Command="npx webpack --mode production --env dev" Condition="'$(Configuration)' == 'DEV'" />
    <Exec Command="npx webpack --mode production --env beta" Condition="'$(Configuration)' == 'BETA'" />
    <Exec Command="npx webpack --mode production --env prod" Condition="'$(Configuration)' == 'PROD'" />
```
Whether the concept was intentional or not, it's now really easy to understand what is going on, and wire the environment through your build pipeline.

HTH! Thanks for working to move this PR forward.

Reply:
Yes I think args is possible to pass in. Has seen some people doing that. Is that not working out for you @robertmclaws ?

Reply:
@evenstensberg Not, it's definitely working. The concern on this part of the thread is that the documentation is inconsistent between sections with how it deals with `env` vs `args`. My change here is to make `env` a single string that specifies the environment name, which brings this page in line with documentation here: https://webpack.js.org/concepts/mode/#mode-production and has the side effect of being much simpler than the approach used throughout the rest of the docs.

I'm suggesting that my change be approved here, and then a separate issue opened to update documentation throughout the rest of the site to use the new `module.exports` signature.

Reply:
That might be a good thing, but people use the env variable for other things too, so that might be a diversion from the rest of the JS ecosystem, don't you think? I'm in for keeping env, but it has to be done similar to the rest of the ecosystem

Reply:
To summarize from what we have so far:
- addition of `args` to the signature is pure gold
- changing `env` to `string` is not that common, please roll back